tooling/bundler/src/bundle/windows/templates/installer.nsi

```diff
@@ -493,63 +496,90 @@ SectionEnd
-      Pop $0
-      ${If} $0 = 0
-        DetailPrint "$(webview2DownloadSuccess)"
-      ${Else}
-        DetailPrint "$(webview2DownloadError)"
-        Abort "$(webview2AbortError)"
+  ${If} $4 == ""
+    ; Webview2 installation
+    ;
+    ; Skip if updating
+    ${If} $UpdateMode <> 1
+      !if "${INSTALLWEBVIEW2MODE}" == "downloadBootstrapper"
+        Delete "$TEMP\MicrosoftEdgeWebview2Setup.exe"
+        DetailPrint "$(webview2Downloading)"
+        NSISdl::download "https://go.microsoft.com/fwlink/p/?LinkId=2124703" "$TEMP\MicrosoftEdgeWebview2Setup.exe"
+        Pop $0
+        ${If} $0 = 0
+          DetailPrint "$(webview2DownloadSuccess)"
+        ${Else}
+          DetailPrint "$(webview2DownloadError)"
+          Abort "$(webview2AbortError)"
+        ${EndIf}
+        StrCpy $6 "$TEMP\MicrosoftEdgeWebview2Setup.exe"
+        Goto install_webview2
+      !endif
+
+      !if "${INSTALLWEBVIEW2MODE}" == "embedBootstrapper"
+        Delete "$TEMP\MicrosoftEdgeWebview2Setup.exe"
+        File "/oname=$TEMP\MicrosoftEdgeWebview2Setup.exe" "${WEBVIEW2BOOTSTRAPPERPATH}"
+        DetailPrint "$(installingWebview2)"
+        StrCpy $6 "$TEMP\MicrosoftEdgeWebview2Setup.exe"
+        Goto install_webview2
+      !endif
+
+      !if "${INSTALLWEBVIEW2MODE}" == "offlineInstaller"
+        Delete "$TEMP\MicrosoftEdgeWebView2RuntimeInstaller.exe"
+        File "/oname=$TEMP\MicrosoftEdgeWebView2RuntimeInstaller.exe" "${WEBVIEW2INSTALLERPATH}"
+        DetailPrint "$(installingWebview2)"
+        StrCpy $6 "$TEMP\MicrosoftEdgeWebView2RuntimeInstaller.exe"
+        Goto install_webview2
+      !endif
+
+      Goto webview2_done
+
+      install_webview2:
+        DetailPrint "$(installingWebview2)"
+        ; $6 holds the path to the webview2 installer
+        ExecWait "$6 ${WEBVIEW2INSTALLERARGS} /install" $1
+        ${If} $1 = 0
+          DetailPrint "$(webview2InstallSuccess)"
+        ${Else}
+          DetailPrint "$(webview2InstallError)"
+          Abort "$(webview2AbortError)"
+        ${EndIf}
+      webview2_done:
+    ${EndIf}
+  ${Else}
+    !if "${MINIMUMWEBVIEW2VERSION}" != ""
+      ${VersionCompare} "${MINIMUMWEBVIEW2VERSION}" "$4" $R0
+      ${If} $R0 = 1
+        DetailPrint "$(installingWebview2)"
+        ${If} ${RunningX64}
+          ReadRegStr $R1 HKLM "SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate" "path"
+        ${Else}
+          ReadRegStr $R1 HKLM "SOFTWARE\Microsoft\EdgeUpdate" "path"
+        ${EndIf}
+        ${If} $R1 == ""
+          ReadRegStr $R1 HKCU "SOFTWARE\Microsoft\EdgeUpdate" "path"
+        ${EndIf}
+        ${If} $R1 != ""
+          ; Chromium updater docs: https://source.chromium.org/chromium/chromium/src/+/main:docs/updater/user_manual.md
+          ; Modified from "HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\Microsoft EdgeWebView\ModifyPath"
+          ExecWait `"$R1" /install appguid=${WEBVIEW2APPGUID}&needsadmin=true` $1
```

Comment:
Just note that I can't get it working without `needsadmin` even on the user install

Reply:
then we shouldn't abort on failure, maybe just leave a warning with a dialog box to continue?

Reply:
Makes sense in that case, I was thinking it should fail because if the version is lower some APIs might not be available (we made wry works on earlier versions so this should be mitigated a lot now)

Do I need to translate the warning to all those languages? Not sure how this is done before

Reply:
well, languages are translated and proof-read by the community, but if you can, maybe use google translate to have initial translation as a starting point

Reply:
I feel like the `Error: Installing WebView2 failed with exit code $1` message might be enough, if you think it's better to get a custom message here, I can work on it tomorrow

Reply:
that's fine with me, let's roll with it for now and improve later if needed